File: programmers/python/로또의_최고_순위와_최저_순위.py
# https://programmers.co.kr/learn/courses/30/lessons/77484?language=python3
def solution(lottos, win_nums):
    answer = []
    
    correct_count = 0
    zero_count = 0
    
    for i in lottos:
        if i == 0:
            zero_count += 1
        elif i in win_nums:
            correct_count += 1
    
    if correct_count + zero_count > 1:
        answer.append(7-(correct_count+zero_count))
    else:
        answer.append(6)
        
    if correct_count > 1:
        answer.append(7-correct_count)
    else:
        answer.append(6)
    
    return answer

File: programmers/java/로또의_최고_순위와_최저_순위.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];

        int zero_count = 0;
        int correct_count = 0;

        for(int i: lottos) {
            if(Arrays.stream(win_nums).anyMatch(n -> n==i))
                correct_count++;
            else if(i == 0)
                zero_count++;
        }

        if(correct_count + zero_count > 1)
            answer[0] = (7-(correct_count+zero_count));
        else
            answer[0] = 6;

        if (correct_count > 1)
            answer[1] = 7-correct_count;
        else
            answer[1] = 6;
        
        return answer;
    }
}